components/style/values/specified/box.rs

```diff
@@ -7,15 +7,194 @@
 use Atom;
 use cssparser::Parser;
 use parser::{Parse, ParserContext};
+#[cfg(feature = "servo")]
+use properties::{longhands, PropertyDeclaration};
 use std::fmt;
 use style_traits::{ParseError, ToCss};
 use values::CustomIdent;
 use values::KeyframesName;
+#[cfg(feature = "servo")]
+use values::computed::Context;
 use values::generics::box_::AnimationIterationCount as GenericAnimationIterationCount;
 use values::generics::box_::VerticalAlign as GenericVerticalAlign;
 use values::specified::{AllowQuirks, Number};
 use values::specified::length::LengthOrPercentage;
 
+#[allow(missing_docs)]
+#[derive(Clone, Copy, Debug, Eq, Hash, MallocSizeOf, Parse, PartialEq, ToComputedValue, ToCss)]
+#[cfg_attr(feature = "servo", derive(Deserialize, Serialize))]
+/// Defines an element’s display type, which consists of
+/// the two basic qualities of how an element generates boxes
+/// <https://drafts.csswg.org/css-display/#propdef-display>
+pub enum Display {
+    Inline, Block, InlineBlock,
+    Table, InlineTable, TableRowGroup, TableHeaderGroup,
+    TableFooterGroup, TableRow, TableColumnGroup,
+    TableColumn, TableCell, TableCaption, ListItem, None,
+    #[parse(aliases = "-webkit-flex")]
+    Flex,
+    #[parse(aliases = "-webkit-inline-flex")]
+    InlineFlex,
+    #[cfg(feature = "gecko")]
```

Comment:
I don't think this works as you think, you still need a `cfg` per variant, let's put them back one per line.

Reply:
Just add them back! I think I have misunderstanding to `attributes` 😓